fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java

```diff
@@ -0,0 +1,234 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.sql.common;
+
+import com.google.common.base.Joiner;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.OrderByElement;
+import com.starrocks.analysis.ParseNode;
+import com.starrocks.analysis.SelectList;
+import com.starrocks.analysis.StatementBase;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CTERelation;
+import com.starrocks.sql.ast.ExceptRelation;
+import com.starrocks.sql.ast.IntersectRelation;
+import com.starrocks.sql.ast.JoinRelation;
+import com.starrocks.sql.ast.QueryRelation;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.Relation;
+import com.starrocks.sql.ast.SelectRelation;
+import com.starrocks.sql.ast.SetOperationRelation;
+import com.starrocks.sql.ast.SubqueryRelation;
+import com.starrocks.sql.ast.TableFunctionRelation;
+import com.starrocks.sql.ast.TableRelation;
+import com.starrocks.sql.ast.UnionRelation;
+import com.starrocks.sql.ast.ValuesRelation;
+import com.starrocks.sql.optimizer.base.SetQualifier;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SqlDigestBuilder {
```

Comment:
Plz add some comment

Reply:
done

Reply:
Can be support a common SqlBuilder, then the SqlDigest, View, ToSql to extend it